x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunnerTests.java

```diff
@@ -1420,7 +1433,83 @@ protected void ensureSingleSegment(String destinationIndexName, int attempt) {
         ensureEnrichIndexIsReadOnly(createdEnrichIndex);
     }
 
+    public void testRunnerCancel() throws Exception {
```

Comment:
I think truly integration testing policy execution cancellation is possible (without avoiding test related failures)?